extract_unique_sensors_01.py

```diff
@@ -86,10 +86,10 @@ def extract_all_sensors(sensors_data, sensor_name):
     assert isinstance(sensor_name, str), "Expected a string."
     assert isinstance(sensors_data, list), "Expected a list."
     candidate_lines = []
-    for sensor_data in sensors_data:
-        if sensor_data.startswith(sensor_name):
-            for candidate_line in sensor_data.split():
-                candidate_lines.append(candidate_line)
+    candidate_lines = [
```

Comment:
:+1: very nice...
**Note:** You do not need to define `candidate_lines = []` in line 88, as that variable will be overwritten by 89 also [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

You can read more on nested list comprehensions here:
- https://spapas.github.io/2016/04/27/python-nested-list-comprehensions/
- https://medium.com/analytics-vidhya/writing-python-nested-list-comprehensions-in-3-steps-75bd9a1b79b6

Reply:
I thought my code would break, but you are correct. It gets overwritten, I just verified. Thank you.